addons/website/static/src/components/wysiwyg_adapter/wysiwyg_adapter.js

```diff
@@ -296,10 +310,22 @@ export class WysiwygAdapterComponent extends Wysiwyg {
             // The onStay/leave callbacks are not passed directly as
             // primaryClick/secondaryClick props, so that closing the dialog
             // with "esc" or the top right cross icon also executes onStay.
-            this.dialogs.add(WebsiteDialog, {
+            const dialogProps = pageLeave ? {
+                body: this.env._t("Would you like to save your changes?"),
+                primaryTitle: "Save",
+                primaryClick: async () => {
+                    await this.save();
+                    leaving = true;
+                },
+                secondaryTitle: "Stay Here",
+                showTertiaryButton: true,
+                tertiaryTitle: "Discard",
+                tertiaryClick: () => leaving = true,
```

Comment:
Sorry I forgot to check that: I think that the titles are not translatable like this, you should set them with `this.env._t(...)`  :wink: